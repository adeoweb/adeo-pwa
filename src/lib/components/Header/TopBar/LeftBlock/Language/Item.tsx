import React, { FunctionComponent } from 'react';
import { TStore } from 'src/lib/types/graphql/Store';
import flagLt from 'src/static/icons/flags/lt.png';
import flagRu from 'src/static/icons/flags/ru.png';
import flagEn from 'src/static/icons/flags/en.png';
import flagLv from 'src/static/icons/flags/lv.png';
import flagEt from 'src/static/icons/flags/et.png';

type TItemProps = {
    item: TStore;
    onSelect: (store: TStore) => void;
};

const FLAGS = new Map()
    .set('en', flagEn)
    .set('et', flagEt)
    .set('lt', flagLt)
    .set('lv', flagLv)
    .set('ru', flagRu);

const Item: FunctionComponent<TItemProps> = ({ item, onSelect }) => {
    const { store_name: storeName, locale } = item;
    const [lang] = locale.split('_');
    let img: JSX.Element | null = null;

    if (FLAGS.has(lang)) {
        img = <img src={FLAGS.get(lang)} alt={locale} />;
    }

    return (
        // eslint-disable-next-line jsx-a11y/anchor-is-valid
        <a href="#" onClick={() => onSelect(item)}>
            {img}
            {storeName}
        </a>
    );
};

export default Item;
