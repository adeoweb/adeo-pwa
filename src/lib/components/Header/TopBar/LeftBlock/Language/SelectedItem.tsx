import React, { FunctionComponent } from 'react';

import { TStore } from 'src/lib/types/graphql/Store';

type TSelectedItemProps = {
    item: TStore;
};

const SelectedItem: FunctionComponent<TSelectedItemProps> = ({
    item: { store_name: storeName }
}) => {
    return <div>{storeName}</div>;
};

export default SelectedItem;
