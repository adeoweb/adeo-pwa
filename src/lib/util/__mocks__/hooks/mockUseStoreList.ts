import { TUseStoreList } from 'src/peregrine/lib/talons/adeoweb/App/useStoreList';

const mock: TUseStoreList = {
    loading: false,
    data: [
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lt_lt',
            id: 1,
            locale: 'lt_LT',
            store_name: 'Lithuanian',
            website_id: 1
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lt_en',
            id: 2,
            locale: 'en_US',
            store_name: 'English',
            website_id: 1
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lt_ru',
            id: 3,
            locale: 'ru_RU',
            store_name: 'Russian',
            website_id: 1
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lv_lv',
            id: 4,
            locale: 'lv_LV',
            store_name: 'Latvia',
            website_id: 2
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lv_en',
            id: 5,
            locale: 'en_US',
            store_name: 'English',
            website_id: 2
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'lv_ru',
            id: 6,
            locale: 'ru_RU',
            store_name: 'Russian',
            website_id: 2
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'ee_ee',
            id: 7,
            locale: 'et_EE',
            store_name: 'Estonians',
            website_id: 3
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'ee_en',
            id: 8,
            locale: 'en_US',
            store_name: 'English',
            website_id: 3
        },
        {
            base_url: 'https://stg.lemona.lt/',
            code: 'ee_ru',
            id: 9,
            locale: 'ru_RU',
            store_name: 'Russian',
            website_id: 3
        }
    ]
};

export default mock;
