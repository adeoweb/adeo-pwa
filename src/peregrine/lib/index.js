// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
