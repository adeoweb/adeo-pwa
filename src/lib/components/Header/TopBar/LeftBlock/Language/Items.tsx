import React, { FunctionComponent } from 'react';
import Item from './Item';
import { TStore } from 'src/lib/types/graphql/Store';

type TItemsProps = {
    items: TStore[];
    onSelect: (item: TStore) => void;
};

const Items: FunctionComponent<TItemsProps> = ({ items, onSelect }) => {
    return (
        <div className="header-menu">
            <ul>
                {items.map((item, index) => (
                    <li key={`${item.store_name}-${index}`}>
                        <Item item={item} onSelect={() => onSelect(item)} />
                    </li>
                ))}
            </ul>
        </div>
    );
};

export default Items;
