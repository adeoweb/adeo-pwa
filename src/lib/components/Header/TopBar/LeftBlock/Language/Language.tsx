import React, { FunctionComponent } from 'react';

import Items from 'src/lib/components/Header/TopBar/LeftBlock/Language/Items';
import { DEFAULT_LANG } from 'src/lib/constants/language';
import { TStore } from 'src/lib/types/graphql/Store';
import { useAppContext } from 'src/peregrine/lib/context/adeoweb/app';
import { useStoreList } from 'src/peregrine/lib/talons/adeoweb/App/useStoreList';

import GET_STORE_LIST from '../../../../../queries/getStoreList.graphql';
import SelectedItem from './SelectedItem';

const getStoreByLang = (
    lang: string,
    stores: TStore[] = []
): TStore | undefined => {
    return stores.find(store => store.locale?.indexOf(lang) === 0);
};

const Language: FunctionComponent = () => {
    const [{ activeStore }, { setActiveStore }] = useAppContext();
    const { loading, error, data } = useStoreList({
        query: GET_STORE_LIST
    });

    if (loading || error || !data || data.length <= 1) {
        return null;
    }

    const selectedItem =
        activeStore || getStoreByLang(DEFAULT_LANG, data) || null;

    if (!selectedItem) {
        return null;
    }

    return (
        <div className="header-dropdown">
            <SelectedItem item={selectedItem} />
            <Items items={data} onSelect={setActiveStore} />
        </div>
    );
};

export default Language;
