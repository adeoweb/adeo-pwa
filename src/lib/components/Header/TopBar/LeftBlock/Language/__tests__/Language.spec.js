import React from 'react';

import { createTestInstance } from '@magento/peregrine';

import mockUseStoreList from 'src/lib/util/__mocks__/hooks/mockUseStoreList';

import Language from '../Language';

jest.mock('src/peregrine/lib/context/adeoweb/app', () => {
    const state = {
        activeStore: null
    };
    const api = {
        setActiveStore: jest.fn()
    };
    const useAppContext = jest.fn(() => [state, api]);

    return { useAppContext };
});

jest.mock('src/peregrine/lib/talons/adeoweb/App/useStoreList', () => {
    return {
        useStoreList: () => mockUseStoreList
    };
});

test('Renders correctly', () => {
    const tree = createTestInstance(<Language />);

    expect(tree.toJSON()).toMatchSnapshot();
});
