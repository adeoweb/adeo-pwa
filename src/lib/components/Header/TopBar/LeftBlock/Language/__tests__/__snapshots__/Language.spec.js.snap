// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="header-dropdown"
>
  <div>
    English
  </div>
  <div
    className="header-menu"
  >
    <ul>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="lt_LT"
            src="test-file-stub"
          />
          Lithuanian
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="en_US"
            src="test-file-stub"
          />
          English
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="ru_RU"
            src="test-file-stub"
          />
          Russian
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="lv_LV"
            src="test-file-stub"
          />
          Latvia
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="en_US"
            src="test-file-stub"
          />
          English
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="ru_RU"
            src="test-file-stub"
          />
          Russian
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="et_EE"
            src="test-file-stub"
          />
          Estonians
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="en_US"
            src="test-file-stub"
          />
          English
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="ru_RU"
            src="test-file-stub"
          />
          Russian
        </a>
      </li>
    </ul>
  </div>
</div>
`;
